galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+
+    @param    SCAs                 Specific SCAs for which the PSF should be loaded.  This can be
+                                   either a single number, a tuple, a NumPy array, or a list.  If
+                                   None, then the PSF will be loaded for all SCAs.  Note that the
+                                   objected that is returned is always a list of the same length,
+                                   regardless of whether `SCAs` is specified or not, but if `SCAs`
+                                   was specified then the list entry will be None for all SCAs that
+                                   were not requested.  [default: None]
+    @param    approximate_struts   Should the routine use an approximate representation of the pupil
+                                   plane, with 6 equally-spaced radial struts, instead of the exact
+                                   representation of the pupil plane?  Setting this parameter to
+                                   True will lead to faster calculations, with a slightly less
+                                   realistic PSFs.  [default: False]
+    @param    n_waves              Number of wavelengths to use for setting up interpolation of the
+                                   chromatic PSF objects, which can lead to much faster image
+                                   rendering.  If None, then no interpolation is used. Note that
+                                   users who want to interpolate can always set up the interpolation
+                                   later on even if they do not do so when calling getPSF().
+                                   [default: None]
+    @param    extra_aberrations    Array of extra aberrations to include in the PSF model, on top of
+                                   those that are part of the WFIRST design.  These should be
+                                   provided in units of waves at the fiducial wavelength of 1293 nm,
+                                   as an array of length 12 with entries 4 through 11 corresponding
+                                   to defocus through spherical aberrations.  [default: None]
+    @param    wavelength_limits    A tuple or list of the blue and red wavelength limits to use for
+                                   interpolating the chromatic object, if `n_waves` is not None.  If
+                                   None, then it uses the blue and red limits of all imaging
+                                   passbands to determine the most inclusive wavelength range
+                                   possible.  But this keyword can be used to reduce the range of
+                                   wavelengths if only one passband (or a subset of passbands) is to
+                                   be used for making the images.
+                                   [default: None]
+    @param    verbose              Should the routine provide information about its progress in
```

Comment:
The more appropriate thing for this purpose would be to take a logger object, so these progress statements can be written to wherever the user wants them, not just print statements written to the screen.  

e.g. RealGalaxy has this option, although I'm noticing now that it seems to be undocumented.  Hmm.  Probably ought to remedy that.